transport.go

```diff
@@ -150,13 +150,31 @@ func (t *transport) handle() {
 	for {
 		select {
 		case req := <-t.requests:
-			// BUG(stevvooe): This is an awful tag allocation procedure.
-			// Replace this with something that let's us allocate tags and
-			// associate data with them, returning to them to a pool when
-			// complete. Such a system would provide a lot of information
-			// about outstanding requests.
-			tags++
-			fcall := newFcall(tags, req.message)
+			var selected Tag
+			if req.message.Type() == Tversion {
+				// Tversion can only use NOTAG.
+				if _, exists := outstanding[NOTAG]; exists {
```

Comment:
Please double check, but I believe that we never have version messages when there is an active transport. Please see https://github.com/docker/go-p9p/blob/master/version.go#L18. When the client is created, we do a [synchronous negotiation](https://github.com/docker/go-p9p/blob/master/client.go#L19), then the transport is created.

All we need to do here is ensure that `NOTAG` is not allocated and is it is encountered, there is a protocol error.